cli/cmd/root.go

```diff
@@ -131,6 +132,43 @@ func validatedPublicAPIClient(retryDeadline time.Time) pb.ApiClient {
 	return hc.PublicAPIClient()
 }
 
+type statOptionsBase struct {
```

Comment:
nice!

consider including `fromNamespace` and `fromResource` as part of `statOptionsBase`, that way you wouldn't need a separate `routesOptions` struct.